src/lib/contextMenuContent.component.ts

```diff
@@ -216,6 +226,62 @@ export class ContextMenuContentComponent implements OnInit, OnDestroy, AfterView
     this.changeDetector.markForCheck();
   }
 
+  @HostListener('document:keydown', ['$event'])
```

Comment:
You can let angular handle a lot of this KeyboardEvent logic for you by using listeners like `document:keydown.ArrowDown` on the relevant functions directly.